### PR TITLE
Add `cost_analysis_enabled` option

### PR DESCRIPTION
**Pull request by artificial-aidan**
## Describe your changes

Add the `cost_analysis_enabled` variable

## Issue number

#528

## Checklist before requesting a review
- [X] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [X] I have executed pre-commit on my machine
- [X] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -20,6 +20,7 @@ resource "azurerm_kubernetes_cluster" "main" {
   resource_group_name                 = data.azurerm_resource_group.main.name
   automatic_channel_upgrade           = var.automatic_channel_upgrade
   azure_policy_enabled                = var.azure_policy_enabled
+  cost_analysis_enabled               = var.cost_analysis_enabled
```
Comment (lonegunmanb) [inline main.tf:23]: According to [the document](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#cost_analysis_enabled):

>The sku_tier must be set to `Standard` or `Premium` to enable this feature. 

Could we add a `precondition` [here](https://github.com/Azure/terraform-azurerm-aks/blob/main/main.tf#L626)?

```hcl
precondition {
  condition         = var.cost_analysis_enabled != true || (var.sku_tier == "Standard" || var.sku_tier == "Premium")
  error_message = "`sku_tier` must be either `Standard` or `Premium` when cost analysis is enabled."
}
```

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @artificial-aidan for opening this pr to us, one review comment.

**Comment by artificial-aidan** [comment]
@microsoft-github-policy-service agree company="Artificial Inc"

**Review by lonegunmanb — APPROVED** [review]
Thanks @artificial-aidan , LGTM!

**Pull request merged**